src/include/error.h

```diff
@@ -19,6 +19,15 @@
 #define	WT_DIAGNOSTIC_YIELD
 #endif
 
+#define	__wt_err(session, error, fmt, ...)				\
+	__wt_err_func(session, error, "%s: %d: " fmt,			\
+	    __func__, __LINE__, __VA_ARGS__)
```

Comment:
I don't think an explicit `fmt` is needed. The `__eventv` API already has equivalent func and line parameters. This could just become:
```
#define __wt_err(session, error, ...)        \
__wt_err_func(session, error, __func__, __LINE__, __VA_ARGS__);
```
With a corresponding change to the parameters in `__wt_err_func`.

Then we wouldn't need to change all the callers.

Reply:
I could have sworn I tried to go down this path, but apparently not! Thank you, @agorrod, I totally missed this. @dgottlieb, sorry to have led you down the wrong path... Here's diff following Alex's point:
```
diff --git a/src/include/error.h b/src/include/error.h
index f94c3d7b8..6cb7ae133 100644
--- a/src/include/error.h
+++ b/src/include/error.h
@@ -19,6 +19,11 @@
 #define        WT_DIAGNOSTIC_YIELD
 #endif

+#define        __wt_err(session, error, ...)                                   \
+       __wt_err_func(session, error, __func__, __LINE__, __VA_ARGS__)
+#define        __wt_errx(session, ...)                                         \
+       __wt_errx_func(session, __func__, __LINE__, __VA_ARGS__)
+
 /* Set "ret" and branch-to-err-label tests. */
 #define        WT_ERR(a) do {                                                  \
        if ((ret = (a)) != 0)                                           \
diff --git a/src/support/err.c b/src/support/err.c
index 0569d0545..3459b6cd1 100644
--- a/src/support/err.c
+++ b/src/support/err.c
@@ -309,13 +309,14 @@ err:              if (fprintf(stderr,
 }

 /*
- * __wt_err --
+ * __wt_err_func --
  *     Report an error.
  */
 void
-__wt_err(WT_SESSION_IMPL *session, int error, const char *fmt, ...)
+__wt_err_func(WT_SESSION_IMPL *session,
+    int error, const char *file_name, int line_number, const char *fmt, ...)
     WT_GCC_FUNC_ATTRIBUTE((cold))
-    WT_GCC_FUNC_ATTRIBUTE((format (printf, 3, 4)))
+    WT_GCC_FUNC_ATTRIBUTE((format (printf, 5, 6)))
     WT_GCC_FUNC_ATTRIBUTE((visibility("default")))
 {
        va_list ap;
@@ -325,18 +326,20 @@ __wt_err(WT_SESSION_IMPL *session, int error, const char *fmt, ...)
         * an error value to return.
         */
        va_start(ap, fmt);
-       WT_IGNORE_RET(__eventv(session, false, error, NULL, 0, fmt, ap));
+       WT_IGNORE_RET(__eventv(session,
+           false, error, file_name, line_number, fmt, ap));
        va_end(ap);
 }

 /*
- * __wt_errx --
+ * __wt_errx_func --
  *     Report an error with no error code.
  */
 void
-__wt_errx(WT_SESSION_IMPL *session, const char *fmt, ...)
+__wt_errx_func(WT_SESSION_IMPL *session,
+    const char *file_name, int line_number, const char *fmt, ...)
     WT_GCC_FUNC_ATTRIBUTE((cold))
-    WT_GCC_FUNC_ATTRIBUTE((format (printf, 2, 3)))
+    WT_GCC_FUNC_ATTRIBUTE((format (printf, 4, 5)))
 {
        va_list ap;

@@ -345,7 +348,8 @@ __wt_errx(WT_SESSION_IMPL *session, const char *fmt, ...)
         * an error value to return.
         */
        va_start(ap, fmt);
-       WT_IGNORE_RET(__eventv(session, false, 0, NULL, 0, fmt, ap));
+       WT_IGNORE_RET(__eventv(session,
+           false, 0, file_name, line_number, fmt, ap));
        va_end(ap);
 }

diff --git a/src/support/scratch.c b/src/support/scratch.c
index 2ead79a1c..6d1bbad9f 100644
--- a/src/support/scratch.c
+++ b/src/support/scratch.c
@@ -400,16 +400,19 @@ __wt_scr_discard(WT_SESSION_IMPL *session)
                if (*bufp == NULL)
                        continue;
                if (F_ISSET(*bufp, WT_ITEM_INUSE))
+#ifdef HAVE_DIAGNOSTIC
                        __wt_errx(session,
                            "scratch buffer allocated and never discarded"
-#ifdef HAVE_DIAGNOSTIC
                            ": %s: %d",
                            session->
                            scratch_track[bufp - session->scratch].file,
                            session->
                            scratch_track[bufp - session->scratch].line
-#endif
                            );
+#else
+                       __wt_errx(session,
+                           "scratch buffer allocated and never discarded");
+#endif
 
                __wt_buf_free(session, *bufp);
                __wt_free(session, *bufp);
```
Dan, maybe it's easiest to drop back to develop, add Alex's change, plus your last commit a256544.